base/reflection.jl

```diff
@@ -233,28 +233,51 @@ function methods(f::ANY)
     end
 end
 
+function visit(f, mt::MethodTable)
```

Comment:
i can implement a variation on `start(mt::MethodTable) = @async visit(produce, mt)` that maintains the current index as a mutable state vector

or an immutable iterator (which would be slightly more expensive, since it needs to recreate a new immutable vector at each step)

do you have a preference for either one?

or i could just define that `methods` always returns an array, which has the benefit of being easier to avoid accidentally mutating as you examine it (e.g. `show(methods(show).cache)`)